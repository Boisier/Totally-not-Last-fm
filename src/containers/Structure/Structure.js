// React
import React, { Component } from 'react'

import './Structure.scss'

import Header from 'scenes/Header/Header'
import Footer from 'scenes/Footer/Footer'

export default class extends Component {
  render = () => (
    <section className='App'>
      <Header />
      <section id="page-structure">
        {this.props.children}
      </section>
      <Footer />
    </section>
  )
}
