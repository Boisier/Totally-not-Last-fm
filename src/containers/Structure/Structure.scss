@import '~style/root';

#page-structure
{
  padding-top: $header-height;
  min-height:calc(100% - #{$header-height} );
}
