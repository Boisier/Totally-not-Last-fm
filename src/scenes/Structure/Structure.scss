#page-structure
{
  display: block;

  width: 800px;
  margin: 30px auto auto auto;
}
