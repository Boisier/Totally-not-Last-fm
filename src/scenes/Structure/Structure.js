// React
import React, { Component } from 'react'

import './Structure.scss'

export default class extends Component {
  render = () => (
    <section id="page-structure">
      {this.props.children}
    </section>
  )
}
