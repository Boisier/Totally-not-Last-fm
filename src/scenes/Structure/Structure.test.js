import React from 'react'
import renderer from 'react-test-renderer'
import { MemoryRouter } from 'react-router-dom'

import Structure from './Structure'

it('Conserve children prop', () => {
  const component = renderer.create(
    <MemoryRouter>
      <Structure>Content</Structure>
    </MemoryRouter>
  ).toJSON()
  expect(component).toMatchSnapshot()
})
