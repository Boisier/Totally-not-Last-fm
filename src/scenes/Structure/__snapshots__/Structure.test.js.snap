// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Conserve children prop 1`] = `
<section
  id="page-structure"
>
  Content
</section>
`;
